enforcer/resources/kubernetes_resource_loader.py

```diff
@@ -0,0 +1,41 @@
+import os
+from typing import List
+
+from enforcer.env_vars import DISCOVERY_MAX_BATCHES, DISCOVERY_BATCH_SIZE
+from kubernetes import client
+from kubernetes.client import V1ReplicaSetList
+from kubernetes import config
+
+from enforcer.model import RsOwner
+
+if os.getenv("KUBERNETES_SERVICE_HOST"):
+    config.load_incluster_config()
+else:
+    config.load_kube_config()
+
+
+class KubernetesResourceLoader:
+
+    @staticmethod
+    def load_replicasets() -> List[RsOwner]:
+        cluster_rs: List[RsOwner] = []
+        continue_ref = None
+        for _ in range(DISCOVERY_MAX_BATCHES):
+            replicasets: V1ReplicaSetList = client.AppsV1Api().list_replica_set_for_all_namespaces(
+                limit=DISCOVERY_BATCH_SIZE, _continue=continue_ref
+            )
+
+            for replicaset in replicasets.items:
+                owner_references = replicaset.metadata.owner_references
```

Comment:
I woiuld log a warning if there is more than one owner reference

Reply:
And if there is, do we only want one with controller: true?

Reply:
I think, unlike pod, replica sets can have only one owner reference.
But I'll add handling for the case it has multiple owners